File: GC.WebSpace/ClientApp/domain/records/ads/adBlank.ts
import { Ad } from "./ad";
import { AdType } from "./adType";

export class AdBlank {
    constructor(
        public id: string | null,
        public type: AdType | null,
        public title: string | null,
        public description: string | null,
        public firstName: string | null,
        public middleName: string | null,
        public lastName: string | null,
        public phoneNumber: string | null,
        public publishDate: Date | null,
        public image: string | null
    ) { }
}

export namespace AdBlank {
    export function empty() {
        return new AdBlank(null, null, null, null, null, null, null, null, null, null);
    }

    export function create(ad: Ad) {
        return new AdBlank(ad.id, ad.type, ad.title, ad.description, ad.firstName, ad.middleName, ad.lastName, ad.phoneNumber,
            ad.publishDate, ad.image);
    }
}

File: GC.Domain/Records/Ads/AdBlank.cs
using GC.Domain.Persons;
using System;

namespace GC.Domain.Records.Ads
{
    public class AdBlank : FromPersonBlank
    {
        public Guid? Id { get; set; }
        public AdType? Type { get; set; }
        public String Title { get; set; }
        public String Description { get; set; }
        public String FirstName { get; set; }
        public String MiddleName { get; set; }
        public String LastName { get; set; }
        public String PhoneNumber { get; set; }
        public DateTime? PublishDate { get; set; }
        public String Image { get; set; }
    }
}
